actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -311,7 +311,7 @@ def session_options=(options)
     def GET
       @env["action_dispatch.request.query_parameters"] ||= Utils.deep_munge(normalize_encode_params(super || {}))
     rescue Rack::Utils::ParameterTypeError, Rack::Utils::InvalidParameterError => e
-      raise ActionController::BadRequest.new(:query, e)
+      raise ActionController::BadRequest, "Invalid query parameters: #{e.message}", e.backtrace
```

Comment:
Yeah, I was wondering if the initializer is a public API too. Maybe we want to introduce a new exception class that always has `#cause`?